### ISSUE TITLE
Diffs mit Spielraum nach oben

### ISSUE THREAD

**Issue by joriki** [CatoTH/antragsgruen#137]
Ich sammle hier mal suboptimale Diffs, die nicht falsch sind und nicht dringend geändert werden müssen aber wenn wir mal dazu kommen verbessert werden sollten.

https://bdk.antragsgruen.de/39/motion/118/amendment/196

**Comment by joriki** [comment]
https://bdk.antragsgruen.de/39/motion/118/amendment/289

Diesen beiden ist gemeinsam, dass sie besser aussehen würden, wenn erst die Einfügung und dann die Streichung käme. Das ist übrigens auch, wie es bei Eingeben des Änderungsantrags im Editor angezeigt wird, wenn man Text auswählt und überschreibt -- insofern wäre es vielleicht sinnvoll, allgemein diese Reihenfolge umzukehren und bei Ersetzungen immer zuerst die Einfügung und dann die Streichung zu zeigen -- damit kann man auch zusammenhängender lesen, wie der Text nach der Änderung lauten würde.

**Comment by joriki** [comment]
Hier ist am Ende unnötig "Bürger*innengesellschaft." gestrichen und eingefügt: https://bdk.antragsgruen.de/39/motion/144/amendment/297

**Comment by CatoTH** [comment]
Die meisten Probleme hier sollten nun behoben sein.

**Comment by joriki** [comment]
Ja, sieht jetzt gut aus. Was denkst zu der Frage, ob die Einfügung vor der Streichung stehen sollte?

**Comment by joriki** [comment]
Die doppelte "Bürger*innengesellschaft" ist inzwischen hierhin umgezogen: 
https://bdk.antragsgruen.de/39/motion/144/amendment/319

**Comment by joriki** [comment]
Hier ist "ist" unnötig gestrichen und eingefügt: https://bdk.antragsgruen.de/39/motion/133/amendment/171

**Comment by CatoTH** [comment]
> Die doppelte "Bürger*innengesellschaft" ist inzwischen hierhin umgezogen:  https://bdk.antragsgruen.de/39/motion/144/amendment/319
Aber der Fehler ist doch jetzt weg, oder?

> Hier ist "ist" unnötig gestrichen und eingefügt: https://bdk.antragsgruen.de/39/motion/133/amendment/171
Ok das ist behoben - auch wenn es jetzt dafür ein expliziter Zeilenumbruch ist, der vorher der Auslöser war...

**Comment by joriki** [comment]
Hier ein Beispiel, wo das Diff ungünstig ist, weil pro Absatz verglichen wird:

https://antraege.gruene.de/bdk41/Wir_sorgen_fuer_bezahlbare_Wohnungen_und_lebenswerte_Kommunen_-12200/406

Bei der Gelegenheit bin ich mal die alten Beispiele durchgegangen:

https://bdk.antragsgruen.de/39/motion/118/amendment/196
Das ursprüngliche Problem ist gelöst, aber jetzt wird am Ende unnötig ein Punkt gestrichen und eingefügt.

https://bdk.antragsgruen.de/39/motion/133/amendment/171
unverändert -- "ist ," wird gestrichen und eingefügt (Du hattest geschrieben, es sei behoben)

https://bdk.antragsgruen.de/39/motion/144/amendment/319
Das ist jetzt in Ordnung.

Offen ist auch noch die Frage, ob die Einfügung vor der Streichung stehen sollte.

**Comment by joriki** [comment]
Hier wird "So fordern wir zum Beispiel" unnötig gestrichen und wieder eingefügt: https://antraege.gruene.de/bdk41/Wir_foerdern_eine_innovative_und_faire_Wirtschaft_-61698/1148

**Comment by joriki** [comment]
Hier wird ein sehr großer Bereich als gestrichen und wieder eingefügt dargestellt: https://antraege.gruene.de/bdk41/Wir_staerken_die_Demokratie_-64428/1610

**Comment by joriki** [comment]
Auch hier unnötige Streichung und Wiedereinfügung: https://antraege.gruene.de/bdk41/Wir_streiten_fuer_Toleranz_Vielfalt_und_Selbstbestimmung_-12397/2364

**Comment by joriki** [comment]
Ein etwas anderer Fall: Hier wird die richtige Menge gestrichen, aber nicht erkannt, dass ganze Sätze eingefügt wurden: https://antraege.gruene.de/bdk41/Welt_im_Blick_-29026/3100

**Comment by CatoTH** [comment]
@joriki Zum letzten Fall (https://antraege.gruene.de/bdk41/Welt_im_Blick_-29026/3100): da sehe ich nur eine Einfügung, gar keine Streichungen - ist das der richtige Link oder versteh ich die Meldung nicht richtig?

**Comment by joriki** [comment]
@CatoTH Sorry, "richtige Menge gestrichen" war falsch -- ich meinte, hier geht es nicht darum, dass zuviel gestrichen wird. Ja, es ist nur eine Einfügung. Das gedoppelte "Es geht um" sollte vorne eingefügt werden, zusammen mit dem Rest des Satzes, statt hinten im darauffolgenden Satz.

**Comment by joriki** [comment]
Hier wieder Einfügung von Gestrichenem (wohl wegen Änderung der Absatzstruktur): https://antraege.gruene.de/bdk41/Wir_staerken_die_Demokratie_-64428/913

**Comment by joriki** [comment]
Ebenso, Einfügung von Gestrichenem wegen Änderung der Absatzstruktur: https://antraege.gruene.de/bdk41/Welt_im_Blick_-29026/363

**Comment by joriki** [comment]
Ähnlich wie in https://github.com/CatoTH/antragsgruen/issues/137#issuecomment-303621117 wird das "Wir" isoliert am Anfang des nächsten Absatzes eingefügt, während das bestehende "Wir" mitten im eingefügten Absatz erscheint: https://github.com/CatoTH/antragsgruen/issues/137#issuecomment-303621117

**Comment by joriki** [comment]
"Roma" wird gestrichen und eingefügt: https://antraege.gruene.de/bdk41/Wir_streiten_fuer_Toleranz_Vielfalt_und_Selbstbestimmung_-12397/2489

**Comment by joriki** [comment]
Noch ein Beispiel (aber jetzt haben wir auch glaub ich schon genug :-) https://antraege.gruene.de/bdk41/Wofuer-wir-Verantwortung-uebernehmen-wollen-17447/3276

**Comment by CatoTH** [comment]
@joriki nach den neuesten Commits in der 3.7er-Version ist der Diff soweit verbessert, dass die drei genannten Anträge so aussehen (würden):
<img width="715" alt="bildschirmfoto 2017-06-17 um 17 51 43" src="https://user-images.githubusercontent.com/533440/27254285-d6bb0952-5385-11e7-9ac0-22373d594654.png">
<img width="694" alt="bildschirmfoto 2017-06-17 um 17 51 51" src="https://user-images.githubusercontent.com/533440/27254286-d6d05154-5385-11e7-9056-0b7bd1eeb8c7.png">
<img width="715" alt="bildschirmfoto 2017-06-17 um 17 51 55" src="https://user-images.githubusercontent.com/533440/27254287-d6deb514-5385-11e7-9fb6-4a037d4c3ee1.png">

Allerdings will ich die neue Version eher nicht mehr für die aktuelle BDK einspielen - das wäre mir dann doch zu sehr "Operation bei offenem Herzen"...
